pkg/policy/api/v3/rule.go

```diff
@@ -0,0 +1,235 @@
+// labels contained in the endpointSelector
+//
+// Each rule is split into an ingress section which contains all rules
+// applicable at ingress, and an egress section applicable at egress. For rule
+// types such as `L4Rule` and `CIDR` which can be applied at both ingress and
+// egress, both ingress and egress side have to either specifically allow the
+// connection or one side has to be omitted.
+//
+// Either ingress, egress, or both can be provided. If both ingress and egress
+// are omitted, the rule has no effect.
+type Rule struct {
+	// IdentitySelector selects all endpoints which should be subject to
+	// this rule. Cannot be empty.
+	EndpointSelector IdentitySelector `json:"endpointSelector"`
+
+	// Ingress is a list of IngressRule which are enforced at ingress.
+	// If omitted or empty, this rule does not apply at ingress.
+	//
+	// +optional
+	Ingress []IngressRule `json:"ingress,omitempty"`
+
+	// Egress is a list of EgressRule which are enforced at egress.
+	// If omitted or empty, this rule does not apply at egress.
+	//
+	// +optional
+	Egress []EgressRule `json:"egress,omitempty"`
+
+	// Labels is a list of optional strings which can be used to
+	// re-identify the rule or to store metadata. It is possible to lookup
+	// or delete strings based on labels. Labels are not required to be
+	// unique, multiple rules can have overlapping or identical labels.
+	//
+	// +optional
+	Labels labels.LabelArray `json:"labels,omitempty"`
+
+	// Description is a free form string, it can be used by the creator of
+	// the rule to store human readable explanation of the purpose of this
+	// rule. Rules cannot be identified by comment.
+	//
+	// +optional
+	Description string `json:"description,omitempty"`
+}
+
+type IngressRule struct {
+	FromIdentities *IdentityRule `json:"fromIdentities,omitempty"`
+
+	FromRequires *EndpointRequire `json:"fromRequires,omitempty"`
+
+	FromCIDRs *CIDRRule `json:"fromCIDR,omitempty"`
+
+	FromEntities *EntityRule `json:"fromEntities,omitempty"`
+}
+
+type EgressRule struct {
+	ToIdentities *IdentityRule `json:"toIdentities,omitempty"`
+
+	ToRequires *EndpointRequire `json:"toRequires,omitempty"`
+
+	ToCIDRs *CIDRRule `json:"toCIDR,omitempty"`
+
+	ToEntities *EntityRule `json:"toEntities,omitempty"`
+
+	ToServices *ServiceRule `json:"toServices,omitempty"`
+}
+
+type IdentityRule struct {
+	IdentitySelector IdentitySelector `json:"identitySelector"`
+	ToPorts          *PortRule        `json:"toPorts,omitempty"`
+}
+
+type EndpointRequire struct {
+	IdentitySelector []IdentitySelector `json:"anyOf"`
+}
+
+type CIDRRule struct {
+	CIDR        []CIDR    `json:"anyOf"`
+	ExceptCIDRs []CIDR    `json:"except,omitempty"`
+	Generated   bool      `json:"-"`
+	ToPorts     *PortRule `json:"toPorts,omitempty"`
+}
+
+type EntityRule struct {
+	Entities []Entity  `json:"anyOf"`
+	ToPorts  *PortRule `json:"toPorts,omitempty"`
+}
+
+// K8sServiceNamespace is an abstraction for the k8s service + namespace types.
```

Comment:
k8s --> Kubernetes

I do not think we should not be using abbreviations in publicly-facing documentation.